examples/gdp/medTermPurchasing_Literal.py

```diff
@@ -1,602 +1,616 @@
+       # only enforce these if we aren't in the last time period
+       if t < model.TimePeriods[-1]:
+           disjunct.amount2 = Constraint(expr=model.AmountPurchased_FD[j, t+1] >= \
+               model.MinAmount_Length[j,2])
+           disjunct.price2 = Constraint(expr=model.Cost_FD[j,t+1] == \
+               model.Prices_Length[j,2,t] * model.AmountPurchased_FD[j, t+1])
+    model.FD_2mo_contract = Disjunct(
+       model.RawMaterials, model.TimePeriods, rule=FD_2mo_contract)
+
+    def FD_3mo_contract(disjunct, j, t):
+       model = disjunct.model()
+       # NOTE: I think there is a mistake in the GAMS file in line 327.
+       # they use the bulk minamount rather than the length one.
+       #I am doing the same here for validation purposes.
+       disjunct.amount1 = Constraint(expr=model.AmountPurchased_FD[j,t] >= \
+           model.MinAmount_Bulk[j,3])
+       disjunct.cost1 = Constraint(expr=model.Cost_FD[j,t] == \
+           model.Prices_Length[j,3,t] * model.AmountPurchased_FD[j,t])
+       # check we aren't in one of the last two time periods
+       if t < model.TimePeriods[-1]:
+           disjunct.amount2 = Constraint(expr=model.AmountPurchased_FD[j,t+1] >= \
+               model.MinAmount_Length[j,3])
+           disjunct.cost2 = Constraint(expr=model.Cost_FD[j,t+1] == \
+               model.Prices_Length[j,3,t] * model.AmountPurchased_FD[j,t+1])
+       if t < model.TimePeriods[-2]:
+           disjunct.amount3 = Constraint(expr=model.AmountPurchased_FD[j,t+2] >= \
+               model.MinAmount_Length[j,3])
+           disjunct.cost3 = Constraint(expr=model.Cost_FD[j,t+2] == \
+               model.Prices_Length[j,3,t] * model.AmountPurchased_FD[j,t+2])
+    model.FD_3mo_contract = Disjunct(
+       model.RawMaterials, model.TimePeriods, rule=FD_3mo_contract)
+
+    def FD_no_contract(disjunct, j, t):
+        model = disjunct.model()
+        disjunct.amount1 = Constraint(expr=model.AmountPurchased_FD[j,t] == 0)
+        disjunct.cost1 = Constraint(expr=model.Cost_FD[j,t] == 0)
+        if t < model.TimePeriods[-1]:
+            disjunct.amount2 = Constraint(expr=model.AmountPurchased_FD[j,t+1] == 0)
+            disjunct.cost2 = Constraint(expr=model.Cost_FD[j,t+1] == 0)
+        if t < model.TimePeriods[-2]:
+            disjunct.amount3 = Constraint(expr=model.AmountPurchased_FD[j,t+2] == 0)
+            disjunct.cost3 = Constraint(expr=model.Cost_FD[j,t+2] == 0)
+    model.FD_no_contract = Disjunct(
+       model.RawMaterials, model.TimePeriods, rule=FD_no_contract)
+
+    def FD_contract(model, j, t):
+       return [ model.FD_1mo_contract[j,t], model.FD_2mo_contract[j,t],
+                model.FD_3mo_contract[j,t], model.FD_no_contract[j,t], ]
+    model.FD_contract = Disjunction(model.RawMaterials, model.TimePeriods,
+       rule=FD_contract)
+
+    return model
+
+
+def build_concrete():
+    return build_model().create_instance('medTermPurchasing_Literal_Chull.dat')
```

Comment:
Why have this function?  It only seems to be used in the `if __name__ == '__main__'` below.